web/src/components/FOI/Home/Redlining.js

```diff
@@ -2933,10 +3052,203 @@ const Redlining = React.forwardRef(
+
+                  pageMappings[doc.documentid][flagInfo.page] =
+                    pageIndex +
+                    totalPageCount -
+                    pagesToRemoveEachDoc.length;
+                } else {
+                  if (flagInfo.flagid !== pageFlagTypes["Consult"]) {
+                    pageMappings[doc.documentid][flagInfo.page] =
+                      pageIndex +
+                      totalPageCount -
+                      pagesToRemoveEachDoc.length;
+                  }
+                }
+
+                // Check if the page has relevant consult flag, if not remove the page
+                let hasConsult = false;
+                for (let consult of doc.consult) {
+                  if (consult.page == flagInfo.page && consult.programareaid.includes(divObj.divisionid)) {
+                    hasConsult = true;
+                    break;
+                  }
+                }
+                if (!hasConsult) {
+                  if (!pagesToRemoveEachDoc.includes(flagInfo.page)) {
+                    pagesToRemoveEachDoc.push(flagInfo.page);
+                    if(!skipDocumentPages) {
+                      delete pageMappings[doc.documentid][flagInfo.page];
+                      pagesToRemove.push(pageIndex + totalPageCountIncludeRemoved)
+                    }
+                  }
+                }
+                if (flagInfo.flagid !== pageFlagTypes["Consult"]) {
+                  pageIndex ++;
+                }
+              }
+            }
+            //End of pageMappingsByConsults
+
+            totalPageCount += Object.keys(
+              pageMappings[doc.documentid]
+            ).length;
+            if (!skipDocumentPages) {
+              totalPageCountIncludeRemoved += doc.pagecount;
+            }
+          }
+        }
+        divPageMappings[divObj.divisionid] = pageMappings;
+        removepages[divObj.divisionid] = pagesToRemove;
+        duplicateWatermarkPages[divObj.divisionid] = duplicateWatermarkPagesEachDiv;
+        NRWatermarksPages[divObj.divisionid] = NRWatermarksPagesEachDiv;
+        pagesToRemove = [];
+        duplicateWatermarkPagesEachDiv = [];
+        NRWatermarksPagesEachDiv = [];
+        totalPageCount = 0;
+        totalPageCountIncludeRemoved = 0;
+        pageMappings = {}
+      }
+
+      setRedlinepageMappings({
+        'divpagemappings': divPageMappings,
+        'pagemapping': pageMappings,
+        'pagestoremove': removepages
+      });
+      setRedlineWatermarkPageMapping({
+        'duplicatewatermark': duplicateWatermarkPages,
+        'NRwatermark': NRWatermarksPages
+      });
+    }
+
     const prepareRedlineIncompatibleMapping = (redlineAPIResponse) => {
       let divIncompatableMapping = {};
       let incompatibleFiles = [];
       let divCounter = 0;
+      if (redlineAPIResponse.consultdocumentlist) {
+        redlineAPIResponse.divdocumentList = redlineAPIResponse.consultdocumentlist
+      }

```

Comment:
This is a bit of a hack, but we are treating consult public bodies as divisions to reuse the divisions logic in a lot of places